velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -404,6 +404,128 @@ TEST_F(DateTimeFunctionsTest, yearTimestampWithTimezone) {
           "year(c0)", std::nullopt, std::nullopt));
 }
 
+TEST_F(DateTimeFunctionsTest, week) {
+  const auto week = [&](std::optional<Timestamp> date) {
+    return evaluateOnce<int64_t>("week(c0)", date);
+  };
+
+  EXPECT_EQ(std::nullopt, week(std::nullopt));
+  EXPECT_EQ(1, week(Timestamp(0, 0)));
+  EXPECT_EQ(53, week(Timestamp(-1, 9000)));
+  EXPECT_EQ(40, week(Timestamp(4000000000, 123000000)));
+  EXPECT_EQ(34, week(Timestamp(998474645, 321000000)));
+
+  setQueryTimeZone("Pacific/Apia");
+  EXPECT_EQ(std::nullopt, week(std::nullopt));
+  EXPECT_EQ(53, week(Timestamp(0, 0)));
+  EXPECT_EQ(53, week(Timestamp(-1, 12300000000)));
+  EXPECT_EQ(40, week(Timestamp(4000000000, 123000000)));
```

Comment:
do we have a `parseTimestamp` like `parseDate`?

Reply:
Thanks for the suggestion, the function ```fromTimeString``` takes input in HH:MM:SS format and is more readable.